frontend/public/components/secrets/create-secret.tsx

```diff
@@ -2,52 +2,77 @@
 import * as _ from 'lodash-es';
 import * as React from 'react';
 import { Helmet } from 'react-helmet';
-import { connect } from 'react-redux';
 import { Link } from 'react-router-dom';
 
 import { k8sCreate, k8sUpdate, K8sResourceKind } from '../../module/k8s';
-import { ButtonBar, Firehose, history, kindObj, StatusBox } from '../utils';
-import { getActiveNamespace, formatNamespacedRouteForResource, UIActions } from '../../ui/ui-actions';
-import { SafetyFirst } from '../safety-first';
+import { ButtonBar, Firehose, history, kindObj, StatusBox, FileInput } from '../utils';
+import { formatNamespacedRouteForResource } from '../../ui/ui-actions';
 import { WebHookSecretKey } from '../secret';
 
-export enum SecretTypes {
-  webhook = 'webhook',
+enum SecretTypeAbstraction {
   generic = 'generic',
+  source = 'source',
+  webhook = 'webhook',
+}
+
+export enum SecretType {
+  basicAuth = 'kubernetes.io/basic-auth',
+  sshAuth = 'kubernetes.io/ssh-auth',
+  opaque = 'Opaque'
 }
 
+const secretFormExplanation = {
+  source: 'Source secrets allow you to authenticate against the SCM server.',
+  webhook: 'Webhook secrets allow you to authenticate a webhook trigger.',
+};
+
+const determineDefaultSecretType = (typeAbstraction) => {
+  return typeAbstraction === 'source' ? SecretType.basicAuth : SecretType.opaque;
+};
+
 const determineSecretTypeAbstraction = (data) => {
-  return _.has(data, WebHookSecretKey) ? SecretTypes.webhook : SecretTypes.generic;
+  return _.has(data, WebHookSecretKey) ? SecretTypeAbstraction.webhook : SecretTypeAbstraction.source;
 };
 
-class BaseEditSecret_ extends SafetyFirst<BaseEditSecretProps_, BaseEditSecretState_> {
+const generateSecret = () => {
+  // http://stackoverflow.com/questions/105034/create-guid-uuid-in-javascript
+  const s4 = () => Math.floor((1 + Math.random()) * 0x10000).toString(16).substring(1);
+  return s4() + s4() + s4() + s4();
+};
+
+// withSecretForm returns SubformComponent which is a Higher Order Component for all the types of secret forms.
+const withSecretForm = (SubformComponent) => class SecretFormComponent extends React.Component<BaseEditSecretProps_, BaseEditSecretState_> {
```

Comment:
I think you can take "Component" out of the name. I'd call this `class SecretForm`

Reply:
Make even more sense ! :)